### ISSUE TITLE
RetryAfterMiddleware missing trait?

### ISSUE THREAD

**Issue by emilbonnek** [melotic/reqwest-retry-after#5]
Pretty new to Rust so apologies if I am holding the tool wrong.
But I am trying to do the minimal example from the docs and I am getting this:

```
error[E0277]: expected a `Fn<(Request, &'a mut task_local_extensions::extensions::Extensions, Next<'a>)>` closure, found `RetryAfterMiddleware`
  --> mylib\src\lib.rs:32:15
   |
32 |         .with(retry_after_middleware)
   |          ---- ^^^^^^^^^^^^^^^^^^^^^^ expected an `Fn<(Request, &'a mut task_local_extensions::extensions::Extensions, Next<'a>)>` closure, found `RetryAfterMiddleware`
   |          |
   |          required by a bound introduced by this call
   |
   = help: the trait `for<'a> Fn<(Request, &'a mut task_local_extensions::extensions::Extensions, Next<'a>)>` is not implemented for `RetryAfterMiddleware`
   = note: required for `RetryAfterMiddleware` to implement `reqwest_middleware::Middleware`
note: required by a bound in `reqwest_middleware::ClientBuilder::with`
  --> C:\Users\username\.cargo\registry\src\index.crates.io-6f17d22bba15001f\reqwest-middleware-0.2.2\src\client.rs:39:12
   |
37 |     pub fn with<M>(self, middleware: M) -> Self
   |            ---- required by a bound in this associated function
38 |     where
39 |         M: Middleware,
   |            ^^^^^^^^^^ required by this bound in `ClientBuilder::with`
```
repro:
```rust
use reqwest_middleware::{ClientBuilder, ClientWithMiddleware};
use reqwest_retry_after::RetryAfterMiddleware;

let client = ClientBuilder::new(reqwest::Client::new())
    .with(RetryAfterMiddleware::new())
    .build();
```

**Comment by evroon** [comment]
I noticed the same problem, seems like the trait cannot be used outside of `lib.rs`. Copying `lib.rs` into your own repo and importing it directly works, but would be nice to import it as dependency instead.

**Comment by melotic** [comment]
Fixed in `0.1.2`. Thanks!
